indico/modules/events/editing/client/js/editing/timeline/ReviewComment.jsx

```diff
@@ -0,0 +1,84 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2023 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import {Form as FinalForm} from 'react-final-form';
+import {Button, Form} from 'semantic-ui-react';
+
+import {FinalSubmitButton, FinalTextArea} from 'indico/react/forms';
+import {Translate} from 'indico/react/i18n';
+
+import {RevisionType} from '../../models';
+
+import {blockPropTypes} from './util';
+
+import './ReviewComment.module.scss';
+
+export default function ReviewComment({block, canEdit}) {
+  const [editFormOpen, setEditFormOpen] = useState(false);
+
+  const handleSubmit = async formData => {
+    console.log(formData);
+    setEditFormOpen(false);
+  };
+
+  console.log(block);
+
+  return (
+    <div styleName="revision-comment">
+      {editFormOpen ? (
+        <div className="f-self-stretch">
+          <FinalForm
+            onSubmit={handleSubmit}
+            initialValues={{text: block.comment}}
+            subscription={{submitting: true}}
+          >
+            {fprops => (
+              <Form onSubmit={fprops.handleSubmit}>
+                <FinalTextArea
+                  name="text"
+                  placeholder={Translate.string('Leave a comment...')}
+                  autoFocus
+                  required={block.type.name !== RevisionType.acceptance}
+                  hideValidationError
+                />
+                <Form.Group styleName="submit-buttons" inline>
+                  <FinalSubmitButton label={Translate.string('Save changes')} />
+                  <Button
+                    disabled={fprops.submitting}
+                    content={Translate.string('Cancel')}
+                    onClick={() => {
+                      fprops.form.reset();
```

Comment:
This shouldn't be necessary - when unmounting the FinalForm any form state should be lost.